### PR TITLE
Handle related pages for SSR

### LINKED ISSUE #43: [SSR Public Catalog] Remove code / automation for manually fetching quickstarts
## Description
Once we have fully swapped [the landing page](https://github.com/newrelic/developer-website/issues/1853) and [the detail page](https://github.com/newrelic/developer-website/issues/1854) over to being powered by Gatsby SSR functions, we no longer need to have a Github workflow that runs every few hours to make API requests and build up a JSON file.

This ticket is to clean up that functionality and the files we are no longer using.

## Acceptance Criteria
* [x] https://github.com/newrelic/developer-website/issues/1853 (landing page via SSR)
* [ ] newrelic/instant-observability-website#42 (search and filter via SSR)
* [x] https://github.com/newrelic/developer-website/issues/1854 (details page via SSR)
* [x] Remove the [Github workflow](https://github.com/newrelic/developer-website/blob/develop/.github/workflows/fetch-quickstarts.yml)
* [x] Remove the [fetch quickstarts script](https://github.com/newrelic/developer-website/blob/develop/scripts/actions/fetch-quickstarts.js) (and related tests)
* [x] Remove the [quickstarts JSON file](https://github.com/newrelic/developer-website/blob/main/src/data/quickstarts.json)
* [x] Remove any code that references the quickstart JSON file ([example](https://github.com/newrelic/developer-website/blob/develop/scripts/list-quickstart-urls.js))
* [ ] Edit the way we give related resources the quickstart slugs (we'll need to make an api call since we don't have quickstarts in the gatsby schema anymore)
* [ ] Remove any Github secrets we no longer need

### PR DESCRIPTION
**Pull request by aswanson-nr**
# Summary
* adds a local plugin to pull in quickstart metadata
* that data can be used to query for related pages for each quickstart
* this replaces the old model because we are no longer pulling down and
  using all the quickstarts at build time

## Before merging
Set the following environment variables for github actions
* `NERDGRAPH_URL`
* `NEW_RELIC_API_KEY`

Closes #43

### CODE REVIEW

File: gatsby-source-quickstart-metadata/gatsby-node.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
+const fetch = require('node-fetch');
+const resolveQuickstartSlug = require('../src/utils/resolveQuickstartSlug');
+const pluginLogPrefix = `[gatsby-source-quickstart-metadata]`;
+const QUICKSTART_NODE_TYPE = `Quickstarts`;
+const NERDGRAPH_URL = process.env.NERDGRAPH_URL;
+const NEW_RELIC_API_KEY = process.env.NEW_RELIC_API_KEY;
+
+exports.onPreInit = () =>
+  console.log(`Loaded gatsby-source-quickstart-metadata`);
```
Comment (nr-kkenney) [inline gatsby-source-quickstart-metadata/gatsby-node.js:9]: would we want the log prefix used here?

File: gatsby-source-quickstart-metadata/gatsby-node.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+        'Api-Key': NEW_RELIC_API_KEY,
+      },
+    });
+
+    if (!resp.ok) {
+      throw Error(
+        `${pluginLogPrefix} Non 200 status code returned from nerdgraph: ${resp.status} ${resp.statusText}`
```
Comment (nr-kkenney) [inline gatsby-source-quickstart-metadata/gatsby-node.js:101]: idt we want to include the log prefix in error messages.

also, these would be caught and shown in the error logging, so we would see 2x the prefix.

File: gatsby-source-quickstart-metadata/gatsby-node.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      );
+    }
+
+    const json = await resp.json();
+
+    if (json.errors) {
+      throw new Error(
+        `${pluginLogPrefix} Errors returned from nerdgraph`,
+        json.errors
```
Comment (nr-kkenney) [inline gatsby-source-quickstart-metadata/gatsby-node.js:110]: same comment as above.

File: gatsby-source-quickstart-metadata/gatsby-node.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      );
+    }
+    const results = json.data?.actor?.nr1Catalog?.quickstarts?.results;
+
+    /* eslint-disable-next-line no-console */
+    console.log(`${pluginLogPrefix} Found ${results?.length} quickstarts`);
```
Comment (nr-kkenney) [inline gatsby-source-quickstart-metadata/gatsby-node.js:116]: i think this log might be slightly better with a number for `results.length`. 

if the previous line was:
```
const results = json.data?.actor?.nr1Catalog?.quickstarts?.results || [ ];
```

but i think as it is right now is fine as well.

**Review by nr-kkenney — CHANGES_REQUESTED** [review]
dropped a couple comments. looks good overall.

File: gatsby-source-quickstart-metadata/gatsby-node.js (hunk 2 above)
Comment (aswanson-nr) [reply gatsby-source-quickstart-metadata/gatsby-node.js:101]: I thought that would be the case, but didn't see gatsby auto prefixing anything during testing. Even the errors.

Comment (aswanson-nr) [reply gatsby-source-quickstart-metadata/gatsby-node.js:101]: ah, I see what you're saying. Will update

File: gatsby-source-quickstart-metadata/gatsby-node.js (hunk 1 above)
Comment (aswanson-nr) [reply gatsby-source-quickstart-metadata/gatsby-node.js]: Good call, updated!

File: gatsby-source-quickstart-metadata/gatsby-node.js (hunk 2 above)
Comment (aswanson-nr) [reply gatsby-source-quickstart-metadata/gatsby-node.js]: Updated!

File: gatsby-source-quickstart-metadata/gatsby-node.js (hunk 3 above)
Comment (aswanson-nr) [reply gatsby-source-quickstart-metadata/gatsby-node.js]: Got it!

File: gatsby-source-quickstart-metadata/gatsby-node.js (hunk 4 above)
Comment (aswanson-nr) [reply gatsby-source-quickstart-metadata/gatsby-node.js]: Good call, updated!

**Review by nr-kkenney — APPROVED** [review]

**Pull request merged**